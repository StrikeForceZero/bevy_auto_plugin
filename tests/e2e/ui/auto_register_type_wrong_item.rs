use bevy_auto_plugin::prelude::*;

#[derive(AutoPlugin)]
#[auto_plugin(impl_plugin_trait)]
struct TestPlugin;

#[auto_register_type(plugin = TestPlugin)]
#[derive(Reflect)]
fn bad_component() {}

// dummy main
fn main() {}
