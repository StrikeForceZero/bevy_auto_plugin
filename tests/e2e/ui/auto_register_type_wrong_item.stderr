error: Attribute macro is not allowed on Fn: Expected Struct or Enum
 --> tests/e2e/ui/auto_register_type_wrong_item.rs:7:1
  |
7 | #[auto_register_type(plugin = TestPlugin)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `auto_register_type` (in Nightly builds, run with -Z macro-backtrace for more info)
